### ISSUE TITLE
mesutBE: Week 2, separation-of-concerns

### ISSUE THREAD

**Issue by MesutBE** [HackYourFutureBelgium/class-8#156]
# Week 2: Step Through

### DevTools Debugger

- [x] [Step-by-Step intro](https://javascript.info/debugging-chrome)
- [x] [the `debugger` statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/debugger)
- [x] [in Chrome/ium](https://developers.google.com/web/tools/chrome-devtools/javascript/)
- [x] [in FireFox](https://www.youtube.com/watch?v=sK8KU8oiF8s)

### Isolating JavaScript

- [x] Type Conversions: [javascript.info/type-conversions](https://javascript.info/type-conversions)
- [x] Operators & Comparisons: [javascript.info/operators](https://javascript.info/operators), [javascript.info/comparison](https://javascript.info/comparison)
- [x] Implicit Coercion: [fcc/news/js-type-coercion](https://www.freecodecamp.org/news/js-type-coercion-explained-27ba3d9a2839/)
- [x] Truthiness: [janke-learning/truthiness](https://github.com/janke-learning/truthiness)
- [x] Loops: [javascript.info/while-for](https://javascript.info/while-for) and complete the tasks

### Integrating JavaScript

- [x] [What are Events in JS?](https://www.youtube.com/watch?v=gx0oAgvXyE4)
- [x] Browser Events: [javascript.info/introduction-browser-events](https://javascript.info/introduction-browser-events)
- [x] UI Events: [javascript.info/mouse-events-basics](https://javascript.info/mouse-events-basics)

#### Isolating JavaScript

* Continue the FreeCodeCamp JavaScript exercises
  * Sign into FreeCodeCamp using your GitHub Account
  * Turn on your Public Profile
  - [ ] Complete [Basic JavaScript](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/) up through [Iterate with JS do ... while loops](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/iterate-with-javascript-do...while-loops)
- [x] Fork [JS Exercises](https://github.com/CodeYourFuture/js-exercises/)
  - [x] Solve the Exercises in the folders called `week-1` and `week-1-practice`
  - [x] You will need to run these exercises [in Node.js](https://www.youtube.com/watch?v=DsH-fizHkcY), but don't be intimidated!  This is actually a very simple way to run your code.  With a little practice using `cd` and `ls` from the command line you'll be good to go!
  - [x] VSC comes built-in with a very good [JavaScript debugger](https://www.youtube.com/watch?v=DsH-fizHkcY) that will help you to step through your code and understand what is happening inside your program.  At first this will be a bit intimidating, but soon you'll wonder how you ever programmed without it.

#### Integrating JavaScript

- [x] Fork [listeners-and-handlers](https://github.com/hackyourfuturebelgium/listeners-and-handlers)
  - [x] Turn on GitHub Pages
  - [x] Push your work regularly, so you can ask questions with issues!
- [x] [handler-refactors](https://github.com/hackyourfuturebelgium/handler-refactors)
  - [x] Turn on GitHub Pages
  - [x]  Push your work regularly, so you can ask questions with issues!
- [x] Fork [using-user-input](https://github.com/hackyourfuturebelgium/using-user-input)
  - [x] Turn on GitHub Pages
  - [ ] Complete `2-fix-the-bugs` and `3-develop-from-specs`
  - [x] Push your work regularly, so you can ask questions with issues!

### Project: [Starter Code](http://hackyourfuture.be/homework-submission/#projects)
- [repository](https://github.com/MesutBE/soc-week-2-project)
- [live](https://mesutbe.github.io/soc-week-2-project/)
- [ ] 1. A README 
- [ ] 1. One branch per file in the `challenges` folder.
- [ ] 1. A live demo with: 0 failing asserts, 5 working buttons

### LAST Exercises
* Continue studying the exercises from in class
  * You can study them live [in the lesson plan](https://hackyourfuture.be/separation-of-concerns/week-2)
  * There are example projects to study in [example-projects](../example-projects).  Don't forget to practice with the Debugger!
  * Feeling motivated? You can clone this repository and try run [the JS exercise source files](../javascript-basics) in Node.js.  Either way, study the exercises in [JS Tutor](http://pythontutor.com/javascript.html#mode=edit) and practice them until you can code them with your eyes closed.

Looking for an extra challenge? Try solving each exercise on a different branch!

---
---
# Useful Links

https://cordova.apache.org/
https://www.electronjs.org/

# Personal Notes
Selection input for color palate
```java script
<input type='color' name='selection' />
```
Different button set
```java script
<input type='button' id='set-color' value='set background' />
```
Reading and process a lot of user inputs > make one form > create new elements > and call them 
```java script
function setColorHandler(event) {
      // read & process user input
      const form = event.target.form;
      const char1 = form.first.value;
      const char2 = form.second.value;
      const char3 = form.third.value;
      const char4 = form.fourth.value;
      const char5 = form.fifth.value;
      const char6 = form.sixth.value;
}
```
---
---

**Comment by greg-hermans** [comment]
Hello Mesut. It seems you didn't complete the week 2 project. You are missing code for the temp-converter, repeater and pad-left exercises.

**Comment by greg-hermans** [comment]
Sorry. I just read that you don't have to finish the project for this assignment. So this is indeed "Done".

**Comment by MesutBE** [comment]
Actually just now I also finished the project @greg-hermans, @colevandersWands 
But I have problems with `temp-converter`.
```java script
// if the user wants to convert to celcius
console.assert(tempConverter(0, 'celcius') === 32, 'fourth');
console.assert(tempConverter(4, 'celcius') === 39.2, 'fifth');
console.assert(tempConverter(12.4, 'celcius') === 54.32, 'sixth');
```
The assert statements are wrong.

They need to be like this I think:

```java script
// if the user wants to convert to celsius
console.assert(tempConverter(89.6, "celsius") === 32, "fourth");
console.assert(tempConverter(102.56, "celsius") === 39.2, "fifth");
console.assert(tempConverter(122.776, "celsius") === 54.32, "sixth");
```
![Screenshot from 2020-02-15 19-56-21](https://user-images.githubusercontent.com/59531743/74593654-c52c8100-502d-11ea-8618-02db085d38a2.png)

![Screenshot from 2020-02-15 20-00-53](https://user-images.githubusercontent.com/59531743/74593668-f2792f00-502d-11ea-9fdd-e2fcbb8ae98c.png)

![Screenshot from 2020-02-15 20-01-15](https://user-images.githubusercontent.com/59531743/74593669-f442f280-502d-11ea-9b5e-64f584e204f8.png)

**Comment by greg-hermans** [comment]
You're right. I think there are some errors in the asserts. You can proceed just as you did. You can also simply use for celsius to farenheit, the reverse asserts as for farenheit to celsius. i.e. : 
// if the user wants to convert to farenheit
console.assert(tempConverter(53, 'farenheit') === 127.4, 'first');
console.assert(tempConverter(0, 'farenheit') === 32, 'second');
console.assert(tempConverter(-40, 'farenheit') === -40, 'third');

// if the user wants to convert to celcius
console.assert(tempConverter(127.4, 'celcius') === 53, 'fourth');
console.assert(tempConverter(32, 'celcius') === 0, 'fifth');
console.assert(tempConverter(-40, 'celcius') === -40, 'sixth');

**Comment by colevandersWands** [comment]
@MesutBE , fixed: https://github.com/HackYourFutureBelgium/soc-week-2-project/blob/master/challenges/temp-converter.js

also fixed the spelling issues 🔠

**Comment by MesutBE** [comment]
Thank you @colevandersWands.
